x-pack/plugins/apm/server/routes/transactions.ts

```diff
@@ -70,17 +71,47 @@ export const transactionGroupsOverviewRoute = createRoute({
       rangeRt,
       uiFiltersRt,
       t.type({
-        size: toNumberRt,
+        transactionType: t.string,
+        latencyAggregationType: latencyAggregationTypeRt,
+      }),
+    ]),
+  }),
+  options: {
+    tags: ['access:apm'],
+  },
+  handler: async ({ context, request }) => {
+    const setup = await setupRequest(context, request);
+
+    const searchAggregatedTransactions = await getSearchAggregatedTransactions(
+      setup
+    );
+
+    const {
+      path: { serviceName },
+      query: { latencyAggregationType, transactionType },
+    } = context.params;
+
+    return getServiceTransactionGroups({
+      setup,
+      serviceName,
+      searchAggregatedTransactions,
+      transactionType,
+      latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+    });
+  },
+});
+
+export const transactionGroupsAggResultsRoute = createRoute({
+  endpoint:
+    'GET /api/apm/services/{serviceName}/transactions/groups/agg_results',
+  params: t.type({
+    path: t.type({ serviceName: t.string }),
+    query: t.intersection([
+      rangeRt,
+      uiFiltersRt,
+      t.type({
+        transactionNames: t.string,
```

Comment:
you could consider using jsonRt here to make sure that the JSON is validated as a request parameter, meaning that invalid JSON will generate a 400 instead of a 500.